MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -1246,3 +1127,129 @@ public final class MagazineLayout: UICollectionViewLayout {
   }
 
 }
+
+// MARK: Layout Attributes Creation and Caching
```

Comment:
it's about time I broke up this file into some sections 😅 this could even be put into its own file (that could also own the cached layout attributes as a private implementation detail). Holding off on that for now to not introduce too many changes. Once these things are stable, it'll be an easy refactor. We could even monitor for things like memory warnings to evict cached layout attributes that haven't been accessed in a while 😄